module/data/advancement/item-grant.mjs

```diff
@@ -0,0 +1,17 @@
+import SpellConfigurationData from "./spell-config.mjs";
+
+export default class ItemGrantConfigurationData extends foundry.abstract.DataModel {
+  static defineSchema() {
+    return {
+      items: new foundry.data.fields.ArrayField(new foundry.data.fields.StringField(), {
+        required: true, label: "DOCUMENT.Items"
+      }),
+      optional: new foundry.data.fields.BooleanField({
+        required: true, label: "DND5E.AdvancementItemGrantOptional", hint: "DND5E.AdvancementItemGrantOptionalHint"
+      }),
+      spell: new foundry.data.fields.EmbeddedDataField(SpellConfigurationData, {
```

Comment:
This same data is going to be used later in the item choice advancement, so I didn't want to have to redefine it.

Reply:
It can still be reusable as a `SchemaField`:
```js
export const SpellConfigurationData = options => new fields.SchemaField({ /* ... */ }, options);
```
Then you can use it like:
```js
spell: SpellConfigurationData({required: true, nullable: true, initial: null})
```

You can also look at `TextureData` for an example of a `SchemaField` that allows some configuration for its inner schema properties.

If you think we're going to be putting logic and functionality on the `SpellConfigurationData` itself, then we can keep it as an `EmbeddedDataField`, but otherwise we should try to be as lean as possible with the data model.

Reply:
I suppose if I left it as a `DataModel` then I could move the `_prepareSpellChanges` method out of `Advancement` and into the data model itself, so it isn't taking space in advancement where it doesn't belong.

Reply:
I've moved the spell changes method from Advancement to SpellConfigurationData which is a nice cleanup :).